kong/pdk/client.lua

```diff
@@ -121,6 +127,75 @@ local function new(self)
   end
 
 
+  ---
+  -- Returns the credentials of the currently authenticated consumer.
+  -- If not set yet, it returns `nil`.
+  -- @function kong.client.get_credential
+  -- @phases access, header_filter, body_filter, log
+  -- @return the authenticated credential
+  -- @usage
+  -- local credential = kong.client.get_credential()
+  -- if credential then
+  --   consumer_id = credential.consumer_id
+  -- else
+  --   -- request not authenticated yet
+  -- end
+  function _CLIENT.get_credential()
+    check_phase(AUTH_AND_LATER)
+
+    return ngx.ctx.authenticated_credential
+  end
+
+
+  ---
+  -- Returns the `consumer` entity of the currently authenticated consumer.
+  -- If not set yet, it returns `nil`.
+  -- @function kong.client.get_consumer
+  -- @phases access, header_filter, body_filter, log
+  -- @treturn table the authenticated consumer entity
+  -- @usage
+  -- local consumer = kong.client.get_consumer()
+  -- if consumer then
+  --   consumer_id = consumer.id
+  -- else
+  --   -- request not authenticated yet, or a credential
+  --   -- without a consumer (external auth)
+  -- end
+  function _CLIENT.get_consumer()
+    check_phase(AUTH_AND_LATER)
+
+    return ngx.ctx.authenticated_consumer
+  end
+
+
+  ---
+  -- Sets the authenticated consumer (and credential) for the current request.
```

Comment:
I think the description also needs to have a note "While both `consumer` and `credential` can be nil, it is required that at least one if them exists. Otherwise this function will throw an error".